packages/peregrine/lib/talons/RootComponents/Product/useProduct.js

```diff
@@ -74,7 +74,9 @@ export const useProduct = props => {
 
         if (data) {
             const productFromNetwork = data.productDetail.items[0];
-            return mapProduct(productFromNetwork);
+            if (productFromNetwork) {
```

Comment:
This is going to be resolved in #2426 but in the conditional. If it gets merged first make sure you switch over so we don't  have an extra check after rebasing :)

Reply:
Good catch, will do!